src/detect-cipservice.c

```diff
@@ -344,14 +344,18 @@ static DetectEnipCommandData *DetectEnipCommandParse(const char *rulestr)
         goto error;
     }
 
-    unsigned long cmd = atol(rulestr);
+    uint64_t cmd;
+    if (ByteExtractStringUint64(&cmd, 10, 0, rulestr) < 0) {
```

Comment:
You should directly use `ByteExtractStringUint16` to avoid further cast

Reply:
Makes sense.